File: src/main/java/ru/hollowhorizon/hc/client/handlers/ClientTickHandler.java
package ru.hollowhorizon.hc.client.handlers;

import net.minecraft.client.Minecraft;
import net.minecraftforge.event.TickEvent;

public class ClientTickHandler {
    public static int ticksInGame = 0;

    public static void clientTickEnd(TickEvent.ClientTickEvent event) {
        if (event.phase == TickEvent.Phase.END) {
            if (!Minecraft.getInstance().isPaused()) {
                ticksInGame++;
            }
        }

//        if (Minecraft.getInstance().level != null) {
//            Minecraft.getInstance().level
//                    .entitiesForRendering()
//                    .forEach(e -> {
//                                if (!(e instanceof IAnimEntity))
//                                    return;
//
//                                ((IAnimEntity)e)
//                                        .getModel()
//                                        .getAnimator()
//                                        .onUpdate();
//                            }
//                    );
//        }
    }
}

File: src/main/java/ru/hollowhorizon/hc/client/handlers/ClientTickHandler.kt
package ru.hollowhorizon.hc.client.handlers

import net.minecraft.client.Minecraft
import net.minecraftforge.event.TickEvent
import net.minecraftforge.event.TickEvent.ClientTickEvent
import net.minecraftforge.event.TickEvent.ServerTickEvent
import ru.hollowhorizon.hc.client.utils.isLogicalClient

object ClientTickHandler {
    @JvmField
    var ticksNotPaused = 0
    @JvmField
    var clientTicks = 0
    @JvmField
    var serverTicks = 0

    @JvmStatic
    fun clientTickEnd(event: ClientTickEvent) {
        if (event.phase == TickEvent.Phase.END) {
            if (!Minecraft.getInstance().isPaused) ticksNotPaused++
            clientTicks++
        }
    }

    @JvmStatic
    fun serverTickEnd(event: ServerTickEvent) {
        if (event.phase == TickEvent.Phase.END) serverTicks++
    }

    fun currentTicks() = if(isLogicalClient) ticksNotPaused else serverTicks
}
